cranelift/wasm/src/code_translator.rs

```diff
@@ -95,6 +97,22 @@ use std::convert::TryFrom;
 use std::vec::Vec;
 use wasmparser::{FuncValidator, MemArg, Operator, WasmModuleResources};
 
+/// Given an `Option<T>`, unwrap the inner `T` or, if the option is `None`, set
+/// the state to unreachable and return.
+///
+/// Used in combination with calling `prepare_addr` and `prepare_atomic_addr`.
```

Comment:
Can we comment *why* we determine that the fallthrough point is unreachable here? (This goes along with the request to document what `None` means below.)